avocado/core/task/statemachine.py

```diff
@@ -79,14 +79,23 @@ async def triage(self):
         except IndexError:
             return
 
-        requirements_ok = await self._spawner.check_task_requirements(runtime_task)
-        if requirements_ok:
-            async with self._state_machine.lock:
-                self._state_machine.ready.append(runtime_task)
-        else:
-            async with self._state_machine.lock:
-                self._state_machine.finished.append(runtime_task)
-                runtime_task.status = 'FAILED ON TRIAGE'
+        # right now, support triage based on requirements only
+        if runtime_task.task.prereqs:
+            finished_tasks_uid = [runtime_task.task.uid for runtime_task
+                                  in self._state_machine.finished]
+            for prereq in runtime_task.task.prereqs:
+                if prereq in finished_tasks_uid:
+                    runtime_task.task.prereqs.remove(prereq)
```

Comment:
This logic is mixing having a prerequisite with a fulfilled prerequisite by considering `prereqs` for both. Can we have two attributes for fulfilled and unfulfilled prerequisites so that we can track both? Alternatively can we have `prereqs` renamed to something less misleading like `blocked_by` or `blockages`?

Reply:
I don't see a reason for not having a list of fulfilled and pending requirements. I'll add that.